### PR TITLE
Add Travis & Doxygen

### PR DESCRIPTION
**Pull request by szymonkaliski**
Notable changes:

- moved processing example into separate `examples_processing` directory for clarity
- we have an issue with this: https://github.com/hoffmannjan/Adafruit_BNO055/blob/master/Adafruit_BNO055.h#L294 - fails on travis-ci on arduino zero but I'm not sure how to "fix" this?

Ping @ladyada :)

### CODE REVIEW

File: Adafruit_BNO055.cpp [hunk 1]
```diff
@@ -1,75 +1,77 @@
...
-/**************************************************************************/
 /*!
-    @brief  Instantiates a new Adafruit_BNO055 class
-*/
-/**************************************************************************/
-Adafruit_BNO055::Adafruit_BNO055(int32_t sensorID, uint8_t address)
-{
+ *  @brief  Instantiates a new Adafruit_BNO055 class
+ *  @param  sensorID
+ *  @param  address
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: i2c address

File: Adafruit_BNO055.cpp [hunk 2]
```diff
@@ -1,75 +1,77 @@
...
-/**************************************************************************/
 /*!
-    @brief  Sets up the HW
-*/
-/**************************************************************************/
-bool Adafruit_BNO055::begin(adafruit_bno055_opmode_t mode)
-{
+ *  @brief  Sets up the HW
+ *  @param  mode
+ *  @return true if process is sucessfull
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: successful

File: Adafruit_BNO055.cpp [hunk 3]
```diff
@@ -1,75 +1,77 @@
...
-
-/**************************************************************************/
 /*!
-    @brief  Sets up the HW
-*/
-/**************************************************************************/
-bool Adafruit_BNO055::begin(adafruit_bno055_opmode_t mode)
-{
+ *  @brief  Sets up the HW
+ *  @param  mode
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: list the valid modes

File: Adafruit_BNO055.cpp [hunk 4]
```diff
@@ -117,25 +110,22 @@ bool Adafruit_BNO055::begin(adafruit_bno055_opmode_t mode)
...
 /*!
-    @brief  Puts the chip in the specified operating mode
-*/
-/**************************************************************************/
-void Adafruit_BNO055::setMode(adafruit_bno055_opmode_t mode)
-{
+ *  @brief  Puts the chip in the specified operating mode
+ *  @param  mode
+ */
+void Adafruit_BNO055::setMode(adafruit_bno055_opmode_t mode) {
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: list the valid modes

File: Adafruit_BNO055.cpp [hunk 5]
```diff
@@ -117,25 +110,22 @@ bool Adafruit_BNO055::begin(adafruit_bno055_opmode_t mode)
...
 
-/**************************************************************************/
 /*!
-    @brief  Changes the chip's axis remap
-*/
-/**************************************************************************/
-void Adafruit_BNO055::setAxisRemap( adafruit_bno055_axis_remap_config_t remapcode )
-{
+ *  @brief  Changes the chip's axis remap
+ *  @param  remapcode
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: list what the remap code is

File: Adafruit_BNO055.cpp [hunk 6]
```diff
@@ -147,13 +137,11 @@ void Adafruit_BNO055::setAxisRemap( adafruit_bno055_axis_remap_config_t remapcod
...
 
-/**************************************************************************/
 /*!
-    @brief  Changes the chip's axis signs
-*/
-/**************************************************************************/
-void Adafruit_BNO055::setAxisSign( adafruit_bno055_axis_remap_sign_t remapsign )
-{
+ *  @brief  Changes the chip's axis signs
+ *  @param  remapsign
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: ditto

File: Adafruit_BNO055.cpp [hunk 7]
```diff
@@ -165,14 +153,11 @@ void Adafruit_BNO055::setAxisSign( adafruit_bno055_axis_remap_sign_t remapsign )
...
-
-/**************************************************************************/
 /*!
-    @brief  Use the external 32.768KHz crystal
-*/
-/**************************************************************************/
-void Adafruit_BNO055::setExtCrystalUse(boolean usextal)
-{
+ *  @brief  Use the external 32.768KHz crystal
+ *  @param  usextal boolean
```
Comment (ladyada) [inline Adafruit_BNO055.cpp]: explain what usextal boolean is used for

**Comment by ladyada** [comment]
didn't go thru the rest of the doxygen comments, but please check all functions and document em best you can :)

**Comment by hoffmannjan** [comment]
now it passes travis :)

**Pull request merged**